mailprocessing/processor/imap.py

```diff
@@ -320,7 +320,7 @@ def _cache_headers_memory(self):
                 self.header_cache[folder]['uidvalidity'] = uidvalidity
                 continue
 
-            if uidvalidity == self.header_cache[folder]['uidvalidity']:
+            if uidvalidity == self.header_cache[folder].get('uidvalidity'):
```

Comment:
`self.header_cache[folder]` might be empty (and not contain `uidvalidity`, if it was created in `set_folders`.